src/sentry/static/sentry/app/utils/withUsersTeams.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+
+import {Team, Organization} from 'app/types';
+import getDisplayName from 'app/utils/getDisplayName';
+import {Client} from 'app/api';
+
+// We require these props when using this HOC
+type DependentProps = {
+  api: Client;
+  organization: Organization;
+};
+
+type InjectedTeamsProps = {
+  teams: Team[];
+  loadingTeams: boolean;
+};
+
+type State = {
+  teams: Team[];
+  loadingTeams: boolean;
+};
+
+const withUsersTeams = <P extends InjectedTeamsProps>(
+  WrappedComponent: React.ComponentType<P>
+) =>
+  class extends React.Component<
+    Omit<P, keyof InjectedTeamsProps> & Partial<InjectedTeamsProps> & DependentProps,
+    State
+  > {
+    static displayName = `withUsersTeams(${getDisplayName(WrappedComponent)})`;
+
+    state = {
+      teams: [],
+      loadingTeams: true,
+    };
+
+    componentDidMount() {
+      this.fetchTeams();
+    }
+
+    async fetchTeams() {
+      this.setState({
+        loadingTeams: true,
+      });
+      const data: Team[] = await this.props.api.requestPromise(
```

Comment:
When this request fails is there an error boundary nearby to catch the exception?

Reply:
Forgot to grab errors will fix in new PR just for creating this HOC